recommender_methods.py

```diff
@@ -99,4 +101,77 @@ def content_recommendation(df, k=20):
             ptid1 = ptid2
             ptid2 = tid
             puid = uid
+    evaluate(pred, actual, k)
     return pred, actual
+
+
+def decay_function(target_time, user_item_time):
+    decay_rate = 0.8
+    return exp(- decay_rate * (target_time - user_item_time))
+
+
+def collaborative_filtering_user_based(rm, df_user_item, user_id, k=2):
+    """
+    performes collaborative_filtering on rating matrix and plot the learning curve.
+    """
+    # Compute peer group of the user
+
+    user_id_row = rm.loc[user_id]
+    with_pearson = rm.corrwith(user_id_row, axis=1)
+
+    sorted_pearson = with_pearson.sort_values(ascending=False)
+    k_closest = sorted_pearson[1: k + 1]
+
+    print("K CLOSEST ", k_closest)
+
+    # Find articles user_id has not read
+    user_articles = df_user_item.loc[user_id]
+
+    user_unread_articles = user_articles[user_articles != 1]
+
+    k_user_read_articles = df_user_item.loc[k_closest.index[0]]
+    k_user_read_articles = k_user_read_articles[k_user_read_articles != 0]
+
+    for i in range(1, len(k_closest)):
```

Comment:
What does the code of this loop do?

Reply:
Den går gjennom k nærmeste brukere å hente alle artikler de har lest. Jeg så at det var noen brukere som ikke hadde overlapp med hva vår bruker ikke hadde lest og hva de brukerne hadde lest. Tenkte at det var heller bedre å returnere en liste med artikler. Vi kan alltid bare velge første artikkel.